newton/_src/usd/utils.py

```
@@ -886,5 +973,592 @@ def _new_vertex_from(v, n_dir):
+        else:
+            color_value = color_input.Get()
+            if color_value is not None:
+                color_np = np.array(color_value, dtype=np.float32)
+                if color_np.size >= 3:
+                    properties["color"] = (float(color_np[0]), float(color_np[1]), float(color_np[2]))
+
+    metallic_input = shader.GetInput("metallic")
+    if metallic_input:
+        try:
+            has_metallic_source = metallic_input.HasConnectedSource()
+        except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)